packages/nimble-components/src/table-column/date-text/cell-view/index.ts

```diff
@@ -20,22 +21,23 @@ export class TableColumnDateTextCellView extends TableColumnTextCellViewBase<
 TableColumnDateTextCellRecord,
 TableColumnDateTextColumnConfig
 > {
-    private static readonly formatter = new Intl.DateTimeFormat(undefined, {
-        dateStyle: 'medium',
-        timeStyle: 'medium'
-    });
+    private formatter?: Intl.DateTimeFormat;
+
+    private columnConfigChanged(): void {
+        this.formatter = createFormatter(this.columnConfig);
+        this.updateText();
+    }
 
     private cellRecordChanged(): void {
-        if (typeof this.cellRecord.value === 'number') {
-            try {
-                this.text = TableColumnDateTextCellView.formatter.format(
-                    this.cellRecord.value
-                );
-            } catch (e) {
-                this.text = '';
-            }
-        } else {
-            this.text = '';
+        this.updateText();
+    }
+
+    private updateText(): void {
+        if (this.formatter) {
```

Comment:
1. If `cellRecordChanged()` runs before `columnConfigChanged()`, then `this.formatter` will be `undefined`. `cellRecord` and `columnConfig` are both bound to equivalent properties on `cellState` in the `TableCell` template. I'm not sure which fires the "changed" handler first.
2. If there is an inconsistency in the configuration of the formatting options, then `this.formatter` will be `undefined`.

In case 2, there will be a validation error, so no cells will render, and `updateText` won't be called. In case 1, I expect that the short-circuiting call to `updateText()` would be followed immediately by one where `this.formatter` was defined. So it should be moot whether we actually set a value for `text` on that first call, because it will be overwritten.

The same applies for the group header view.